.github/workflows/ci.yaml

```diff
@@ -0,0 +1,33 @@
+name: CI
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches: [ main ]
+  push:
+    branches: [ main ]
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  pre-commit:
+    uses: ./.github/workflows/step_pre-commit.yaml
+
+  tests:
+    needs: [ pre-commit ]
+    uses: ./.github/workflows/step_test.yaml
```

Comment:
The goal of these reusable CI is that they could be imported by other projects as well, assuming they have a similar structure, i.e. each project would only define the `ci.yaml` file and import the step workflows from (this) central project. I am planning to make one for `executablebooks` as well so they can have a centralized CI to manage.